pkg/machine/volumes.go

```diff
@@ -29,14 +30,16 @@ func (v VirtIoFs) Kind() string {
 	return string(VirtIOFsVk)
 }
 
-// unitName is the fq path where /'s are replaced with -'s
-func (v VirtIoFs) unitName() string {
-	// delete the leading -
-	unit := strings.ReplaceAll(v.Target, "/", "-")
-	if strings.HasPrefix(unit, "-") {
-		return unit[1:]
+// generateTag generates a tag for VirtIOFs mounts.
+// AppleHV requires tags to be less than 36 bytes
+// Truncate tag to 27 bytes, then add random 8 byte string to avoid collisions
+func (v VirtIoFs) generateTag() string {
+	base := filepath.Base(v.Target)
+	if len(base) > 27 {
+		base = base[:27]
```

Comment:
Why use the base path name at all? Could there be characters that are not allowed?
It seems safer to just use a arbitrary id for everything.

Reply:
Concur. Maybe SHA256 the path and truncate?